src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -196,16 +197,35 @@ private ConvenienceMethod BuildConvenienceMethod(ReturnTypeChain returnTypeChain
                 name = _restClientMethod.Name.IsLastWordSingular() ? $"{_restClientMethod.Name}Value" : $"{_restClientMethod.Name.LastWordToSingular()}Values";
             }
 
-            var parameters = _orderedParameters.Select(p => p.Convenience).WhereNotNull().ToArray();
+            var parameterList = new List<Parameter>();
+            foreach (var parameterChain in _orderedParameters)
+            {
+                if (parameterChain.Input?.Kind == InputOperationParameterKind.Flatten)
+                {
+                    InputType type = parameterChain.Input.Type;
+                    if (type is InputModelType modelType)
+                    {
+                        foreach (var prop in modelType.Properties)
+                        {
+                            var parameter = Parameter.FromModelProperty(prop, prop.Name.ToVariableName(), _typeFactory.CreateType(prop.Type));
+                            parameterList.Add(parameter);
+                        }
+                    }
+
+                } else if (parameterChain.Convenience != null)
+                {
+                    parameterList.Add(parameterChain.Convenience);
+                }
+            }
             var attributes = Operation.Deprecated is { } deprecated
                 ? new[] { new CSharpAttribute(typeof(ObsoleteAttribute), deprecated) }
                 : null;
             var protocolToConvenience = _orderedParameters
                 .Where(p => p.Protocol != null)
-                .Select(p => (p.Protocol!, p.Convenience))
+                .Select(p => (p.Protocol!, p.Convenience, p.Input))
                 .ToArray();
-
-            var convenienceSignature = new MethodSignature(name, _restClientMethod.Summary, _restClientMethod.Description, _restClientMethod.Accessibility | Virtual, returnTypeChain.Convenience, null, parameters, attributes);
+            //var parameters = parameterList.ToArray();
+            var convenienceSignature = new MethodSignature(name, _restClientMethod.Summary, _restClientMethod.Description, _restClientMethod.Accessibility | Virtual, returnTypeChain.Convenience, null, parameterList.ToArray(), attributes);
```

Comment:
ToArray here is kind of redundant because List implements IReadOnlyList
```suggestion
            var convenienceSignature = new MethodSignature(name, _restClientMethod.Summary, _restClientMethod.Description, _restClientMethod.Accessibility | Virtual, returnTypeChain.Convenience, null, parameterList, attributes);
```